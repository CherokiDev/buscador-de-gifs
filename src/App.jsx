import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const App = () => {
  const [categories, setCategories] = useState([""]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories]);
    // setCategories((cat) => [newCategory, ...categories]);
  };

  return (
    <>
      <h1>Buscador de Gifs (prueba)</h1>

      <AddCategory onNewCategory={(value) => onAddCategory(value)} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
