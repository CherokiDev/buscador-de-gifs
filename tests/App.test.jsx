import { render } from "@testing-library/react";
import { App } from "../src/App";

describe("pruebas sobre el componente App", () => {
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
