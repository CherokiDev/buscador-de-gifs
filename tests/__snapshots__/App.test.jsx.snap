// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el componente App Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Buscador de Gifs
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <div
    class="card-grid"
  >
    <p>
      Loading...
    </p>
  </div>
</div>
`;
